articles/azure-monitor/app/tutorial-asp-net-core.md

```diff
@@ -0,0 +1,525 @@
+    ```
+
+4. Immediately following the `var app = builder.Build()` line of code, append the following to configure adaptive sampling rates.
+
+    ```csharp
+    var telemetryConfig = app.Services.GetService<TelemetryConfiguration>();
+    if(telemetryConfig != null)
+    {
+        var telemetryBuilder = telemetryConfig.DefaultTelemetrySink.TelemetryProcessorChainBuilder;
+        telemetryBuilder.UseAdaptiveSampling(maxTelemetryItemsPerSecond:5);    
+        telemetryBuilder.Build();
+    }
+    ```
+
+#### Configure Fixed-rate sampling
+
+Fixed-rate sampling reduces the traffic sent from your web server and web browsers. Unlike adaptive sampling, it reduces telemetry at a fixed rate decided by you.
+
+Like other sampling techniques, this also retains related items. It also synchronizes the client and server sampling so that related items are retained - for example, when you look at a page view in Search, you can find its related server requests.
+
+In Metrics Explorer, rates such as request and exception counts are multiplied by a factor to compensate for the sampling rate, so that they are approximately correct.
+
+1. In Visual Studio 2022, locate and open `Program.cs`.
+
+2. [**Disable adaptive sampling**](#disabling-adaptive-sampling).
+
+3. Enable the fixed-rate sampling module by adding the following code following the `var app = builder.Build();` line of code.
+
+    ```csharp
+    var telemetryConfig = app.Services.GetService<TelemetryConfiguration>();
+    if(telemetryConfig != null)
+    {
+        var telemetryBuilder = telemetryConfig.DefaultTelemetrySink.TelemetryProcessorChainBuilder;
+        double fixedSamplingPercentage = 10;
+        telemetryBuilder.UseSampling(fixedSamplingPercentage);
+        telemetryBuilder.Build();
+    }
+    ```
+
+## Application logging with Application Insights
+
+### Logging overview
+
+Application Insights is one type of logging provider available to ASP.NET Core applications that becomes available to applications when the [Application Insights for ASP.NET Core](#install-the-application-insights-nuget-package) NuGet package is installed and [server-side telemetry collection enabled](#enable-application-insights-server-side-telemetry). As a reminder, the following code in **Program.cs** registers the `ApplicationInsightsLoggerProvider` with the built-in dependency injection container.
+
+```csharp
+builder.Services.AddApplicationInsightsTelemetry();
+```
+
+With the `ApplicationInsightsLoggerProvider` registered as the logging provider, the app is ready to log to Application Insights using either constructor injection with <xref:Microsoft.Extensions.Logging.ILogger> or the generic-type alternative <xref:Microsoft.Extensions.Logging.ILogger%601>. With default settings, the logging provider is configured to automatically capture log events with a severity of <xref:Microsoft.Extensions.Logging.LogLevel.Warning?displayProperty=nameWithType> or greater.
```

Comment:
any way to highlight that the default is to capture Warning or above only?

Reply:
Fixed.